integration/test/performance_tracker/performance_query_utils.go

```diff
@@ -0,0 +1,78 @@
+package data_collector
+
+import (
+	"time"
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+const (
+	Namespace = "CWAgent"
+	DimensionName = "InstanceId"
+	Stat = "Average"
+	Period = 30
+)
+
+// CWGetMetricDataAPI defines the interface for the GetMetricData function
+type CWGetMetricDataAPI interface {
+	GetMetricData(ctx context.Context, params *cloudwatch.GetMetricDataInput, optFns ...func(*cloudwatch.Options)) (*cloudwatch.GetMetricDataOutput, error)
+}
+
+// GetMetrics Fetches the cloudwatch metrics for your provided input in the given time-frame
+func GetMetrics(c context.Context, api CWGetMetricDataAPI, input *cloudwatch.GetMetricDataInput) (*cloudwatch.GetMetricDataOutput, error) {
+	return api.GetMetricData(c, input)
+}
+
+// GenerateGetMetricInputStruct generates the struct required to make a query request to cloudwatch's GetMetrics
+func GenerateGetMetricInputStruct(ids []string, metricNames []string, instanceId string, timeDiff int) (*cloudwatch.GetMetricDataInput, error) {
+	if len(ids) != len(metricNames) {
+		return nil, errors.New("Mismatching lengths of metric ids and metricNames")
+	}
+	
+	if len(ids) == 0 || len(metricNames) == 0 || instanceId == "" || timeDiff == 0 {
+		return nil, errors.New("Must supply metric ids, metric names, instance id, and time to collect metrics")
+	}
+
+	dimensionValue := instanceId
+	metricDataQueries := []types.MetricDataQuery{}
+	
+	//generate list of individual metric requests
+	for i := 0; i < len(ids); i++ {
+		metricDataQueries = append(metricDataQueries, ConstructMetricDataQuery(ids[i], Namespace, DimensionName, dimensionValue, metricNames[i], timeDiff))
+	}
```

Comment:
The `range` keyword returns the index and the value so we can change the for loop here, though I guess I'm not too picky in this scenario since we need to use the index twice.
```suggestion
	for i, id := range ids {
		metricDataQueries = append(metricDataQueries, ConstructMetricDataQuery(id, Namespace, DimensionName, dimensionValue, metricNames[i], timeDiff))
	}
```